wfe/wfe.go

```diff
@@ -1985,13 +2150,39 @@ func (wfe *WebFrontEndImpl) FinalizeOrder(
 	// Lock and update the order with the parsed CSR and the began processing
 	// state.
 	existingOrder.Lock()
+	// do a check on the stored order to see if another process
+	// changed the order status
 	existingOrder.ParsedCSR = parsedCSR
 	existingOrder.BeganProcessing = true
 	existingOrder.Unlock()
 
-	// Ask the CA to complete the order in a separate goroutine.
 	wfe.log.Printf("Order %s is fully authorized. Processing finalization", orderID)
-	go wfe.ca.CompleteOrder(existingOrder)
+
+	// Perform asynchronous finalization
+	go func() {
+		wfe.ca.CompleteOrder(existingOrder)
+
+		// Store the order in this table so we can check if it had previously been replaced.
+		err := wfe.db.AddOrderByIssuedSerial(existingOrder)
+		if err != nil {
+			wfe.sendError(acme.InternalErrorProblem(fmt.Sprintf("Error saving order: %s", err)), response)
```

Comment:
We can't wfe.sendError from an async goroutine, since the HTTP request has already been returned. We can only log. Same on two lines below.